### ISSUE TITLE
Proposal: kill the named arguments in transaction object

### ISSUE THREAD

**Issue by talkol** [orbs-network/orbs-spec#87]
Argument names add at least 4 bytes per argument
It's expensive and increases transaction size
Most languages (like Go) don't support this feature anyways

**Comment by netoneko** [comment]
I disagree because it might be easier to use the smart contract APIs if we have named arguments and not just argument order.

**Comment by talkol** [comment]
From a product perspective it's definitely good, but I'm not sure it's worth the cost. The cost is pretty huge, it increases transaction size significantly

**Comment by OdedWx** [comment]
I agree on that there's a value from a product perspective.

An alternative method to support function overloading is to allow different function signatures to the same function name. with a default signature i.e.:
Deploy: func(a int, b string) 
Call: func(1, "abc")  
Call: func(a)(1) - assigns default b.

This allows to pay for the overhead only when needed. 
BTW in Ethereum you need to support any signature variation in advance as it translates to a different hash.

We can also consider to add non-variadics params as an alternative.

**Comment by OdedWx** [comment]
Related issue: explicit argument types vs blobs (list of byte), taking the argument type from the function signature.
